### PR TITLE
Feature/rails3 fix planning element api

### PR DESCRIPTION
**Pull request by tessi**
- when reading and writing planning_elements fields were named inconsistently
- the name attribute in planning_element_journals was rendered as a namespace, instead as a name attribute
- times should be in iso8601, but they weren't
  `Time.now.utc #=> 2013-10-07 14:39:32 UTC` <-- **THIS IS NOT ISO8601** <sup>(2013-10-07 14:39:32Z is right)</sup>
  
  
- do not listen for `planning_element_type_id` anymore, it is now named `type_id`

see https://www.openproject.org/issues/2297

### CODE REVIEW

**Comment by NobodysNightmare** [comment]
@tessi and me worked on those changes. However we think this could be a highly controversial PR, because there were incompatible changes between **APIv2** _before_ the merge of planning_elements into work_packages and **APIv2** _after_ the merge.

Those changes existed before this PR, however this PR makes the changes consistent instead of reverting them.

There are two ways out:
1. **The hard way**: Repair the API to be consistent with the old behavior
2. **The easy way**: Merge this Pull Request

**Comment by tessi** [comment]
### Now discuss!

**Comment by apfelfabrik** [comment]
Did you test this w/ timelines?

**Comment by apfelfabrik** [comment]
Why is there no issue for this?

**Comment by apfelfabrik** [comment]
And what exactly is being fixed here?

**Comment by NobodysNightmare** [comment]
1. no
2. [description updated](https://www.openproject.org/issues/2297)
3. API was made consistent, plus there was an error preventing updates to the planning-element-type

**Comment by machisuji** [comment]
+1

File: app/models/permitted_params.rb [hunk 1]
```diff
@@ -224,7 +224,7 @@ def self.permitted_attributes
                                                       :start_date,
                                                       :due_date,
                                                       :note,
-                                                      :planning_element_type_id,
+                                                      :type_id,
                                                       :planning_element_status_comment,
                                                       :planning_element_status_id,
```
Comment (mugwump) [inline app/models/permitted_params.rb:229]: :planning_element_status_comment, :planning_element_status_id are also not used anymore and can be deleted.

**Pull request merged**